// osmojs/dist/codegen/osmosis/pool-incentives/v1beta1/shared.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalancerToConcentratedPoolLink = exports.MigrationRecords = void 0;
const binary_1 = require("../../../binary");
function createBaseMigrationRecords() {
    return {
        balancerToConcentratedPoolLinks: []
    };
}
exports.MigrationRecords = {
    typeUrl: "/osmosis.poolincentives.v1beta1.MigrationRecords",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        for (const v of message.balancerToConcentratedPoolLinks) {
            exports.BalancerToConcentratedPoolLink.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMigrationRecords();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.balancerToConcentratedPoolLinks.push(exports.BalancerToConcentratedPoolLink.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMigrationRecords();
        message.balancerToConcentratedPoolLinks = object.balancerToConcentratedPoolLinks?.map(e => exports.BalancerToConcentratedPoolLink.fromPartial(e)) || [];
        return message;
    },
    fromAmino(object) {
        return {
            balancerToConcentratedPoolLinks: Array.isArray(object?.balancer_to_concentrated_pool_links) ? object.balancer_to_concentrated_pool_links.map((e) => exports.BalancerToConcentratedPoolLink.fromAmino(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        if (message.balancerToConcentratedPoolLinks) {
            obj.balancer_to_concentrated_pool_links = message.balancerToConcentratedPoolLinks.map(e => e ? exports.BalancerToConcentratedPoolLink.toAmino(e) : undefined);
        }
        else {
            obj.balancer_to_concentrated_pool_links = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MigrationRecords.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/poolincentives/migration-records",
            value: exports.MigrationRecords.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MigrationRecords.decode(message.value);
    },
    toProto(message) {
        return exports.MigrationRecords.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.poolincentives.v1beta1.MigrationRecords",
            value: exports.MigrationRecords.encode(message).finish()
        };
    }
};
function createBaseBalancerToConcentratedPoolLink() {
    return {
        balancerPoolId: BigInt(0),
        clPoolId: BigInt(0)
    };
}
exports.BalancerToConcentratedPoolLink = {
    typeUrl: "/osmosis.poolincentives.v1beta1.BalancerToConcentratedPoolLink",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.balancerPoolId !== BigInt(0)) {
            writer.uint32(8).uint64(message.balancerPoolId);
        }
        if (message.clPoolId !== BigInt(0)) {
            writer.uint32(16).uint64(message.clPoolId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBalancerToConcentratedPoolLink();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.balancerPoolId = reader.uint64();
                    break;
                case 2:
                    message.clPoolId = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseBalancerToConcentratedPoolLink();
        message.balancerPoolId = object.balancerPoolId !== undefined && object.balancerPoolId !== null ? BigInt(object.balancerPoolId.toString()) : BigInt(0);
        message.clPoolId = object.clPoolId !== undefined && object.clPoolId !== null ? BigInt(object.clPoolId.toString()) : BigInt(0);
        return message;
    },
    fromAmino(object) {
        return {
            balancerPoolId: BigInt(object.balancer_pool_id),
            clPoolId: BigInt(object.cl_pool_id)
        };
    },
    toAmino(message) {
        const obj = {};
        obj.balancer_pool_id = message.balancerPoolId ? message.balancerPoolId.toString() : undefined;
        obj.cl_pool_id = message.clPoolId ? message.clPoolId.toString() : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.BalancerToConcentratedPoolLink.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/poolincentives/balancer-to-concentrated-pool-link",
            value: exports.BalancerToConcentratedPoolLink.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.BalancerToConcentratedPoolLink.decode(message.value);
    },
    toProto(message) {
        return exports.BalancerToConcentratedPoolLink.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.poolincentives.v1beta1.BalancerToConcentratedPoolLink",
            value: exports.BalancerToConcentratedPoolLink.encode(message).finish()
        };
    }
};
