// osmojs/dist/codegen/osmosis/protorev/v1beta1/protorev.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseDenom = exports.PoolWeights = exports.RouteStatistics = exports.Trade = exports.Route = exports.TokenPairArbRoutes = void 0;
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
const binary_1 = require("../../../binary");
function createBaseTokenPairArbRoutes() {
    return {
        arbRoutes: [],
        tokenIn: "",
        tokenOut: ""
    };
}
exports.TokenPairArbRoutes = {
    typeUrl: "/osmosis.protorev.v1beta1.TokenPairArbRoutes",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        for (const v of message.arbRoutes) {
            exports.Route.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.tokenIn !== "") {
            writer.uint32(18).string(message.tokenIn);
        }
        if (message.tokenOut !== "") {
            writer.uint32(26).string(message.tokenOut);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTokenPairArbRoutes();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.arbRoutes.push(exports.Route.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.tokenIn = reader.string();
                    break;
                case 3:
                    message.tokenOut = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseTokenPairArbRoutes();
        message.arbRoutes = object.arbRoutes?.map(e => exports.Route.fromPartial(e)) || [];
        message.tokenIn = object.tokenIn ?? "";
        message.tokenOut = object.tokenOut ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            arbRoutes: Array.isArray(object?.arb_routes) ? object.arb_routes.map((e) => exports.Route.fromAmino(e)) : [],
            tokenIn: object.token_in,
            tokenOut: object.token_out
        };
    },
    toAmino(message) {
        const obj = {};
        if (message.arbRoutes) {
            obj.arb_routes = message.arbRoutes.map(e => e ? exports.Route.toAmino(e) : undefined);
        }
        else {
            obj.arb_routes = [];
        }
        obj.token_in = message.tokenIn;
        obj.token_out = message.tokenOut;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.TokenPairArbRoutes.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/token-pair-arb-routes",
            value: exports.TokenPairArbRoutes.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.TokenPairArbRoutes.decode(message.value);
    },
    toProto(message) {
        return exports.TokenPairArbRoutes.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.TokenPairArbRoutes",
            value: exports.TokenPairArbRoutes.encode(message).finish()
        };
    }
};
function createBaseRoute() {
    return {
        trades: [],
        stepSize: ""
    };
}
exports.Route = {
    typeUrl: "/osmosis.protorev.v1beta1.Route",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        for (const v of message.trades) {
            exports.Trade.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.stepSize !== "") {
            writer.uint32(18).string(message.stepSize);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRoute();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.trades.push(exports.Trade.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.stepSize = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseRoute();
        message.trades = object.trades?.map(e => exports.Trade.fromPartial(e)) || [];
        message.stepSize = object.stepSize ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            trades: Array.isArray(object?.trades) ? object.trades.map((e) => exports.Trade.fromAmino(e)) : [],
            stepSize: object.step_size
        };
    },
    toAmino(message) {
        const obj = {};
        if (message.trades) {
            obj.trades = message.trades.map(e => e ? exports.Trade.toAmino(e) : undefined);
        }
        else {
            obj.trades = [];
        }
        obj.step_size = message.stepSize;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.Route.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/route",
            value: exports.Route.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.Route.decode(message.value);
    },
    toProto(message) {
        return exports.Route.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.Route",
            value: exports.Route.encode(message).finish()
        };
    }
};
function createBaseTrade() {
    return {
        pool: BigInt(0),
        tokenIn: "",
        tokenOut: ""
    };
}
exports.Trade = {
    typeUrl: "/osmosis.protorev.v1beta1.Trade",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.pool !== BigInt(0)) {
            writer.uint32(8).uint64(message.pool);
        }
        if (message.tokenIn !== "") {
            writer.uint32(18).string(message.tokenIn);
        }
        if (message.tokenOut !== "") {
            writer.uint32(26).string(message.tokenOut);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTrade();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.pool = reader.uint64();
                    break;
                case 2:
                    message.tokenIn = reader.string();
                    break;
                case 3:
                    message.tokenOut = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseTrade();
        message.pool = object.pool !== undefined && object.pool !== null ? BigInt(object.pool.toString()) : BigInt(0);
        message.tokenIn = object.tokenIn ?? "";
        message.tokenOut = object.tokenOut ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            pool: BigInt(object.pool),
            tokenIn: object.token_in,
            tokenOut: object.token_out
        };
    },
    toAmino(message) {
        const obj = {};
        obj.pool = message.pool ? message.pool.toString() : undefined;
        obj.token_in = message.tokenIn;
        obj.token_out = message.tokenOut;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.Trade.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/trade",
            value: exports.Trade.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.Trade.decode(message.value);
    },
    toProto(message) {
        return exports.Trade.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.Trade",
            value: exports.Trade.encode(message).finish()
        };
    }
};
function createBaseRouteStatistics() {
    return {
        profits: [],
        numberOfTrades: "",
        route: []
    };
}
exports.RouteStatistics = {
    typeUrl: "/osmosis.protorev.v1beta1.RouteStatistics",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        for (const v of message.profits) {
            coin_1.Coin.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.numberOfTrades !== "") {
            writer.uint32(18).string(message.numberOfTrades);
        }
        writer.uint32(26).fork();
        for (const v of message.route) {
            writer.uint64(v);
        }
        writer.ldelim();
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRouteStatistics();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.profits.push(coin_1.Coin.decode(reader, reader.uint32()));
                    break;
                case 2:
                    message.numberOfTrades = reader.string();
                    break;
                case 3:
                    if ((tag & 7) === 2) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.route.push(reader.uint64());
                        }
                    }
                    else {
                        message.route.push(reader.uint64());
                    }
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseRouteStatistics();
        message.profits = object.profits?.map(e => coin_1.Coin.fromPartial(e)) || [];
        message.numberOfTrades = object.numberOfTrades ?? "";
        message.route = object.route?.map(e => BigInt(e.toString())) || [];
        return message;
    },
    fromAmino(object) {
        return {
            profits: Array.isArray(object?.profits) ? object.profits.map((e) => coin_1.Coin.fromAmino(e)) : [],
            numberOfTrades: object.number_of_trades,
            route: Array.isArray(object?.route) ? object.route.map((e) => BigInt(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        if (message.profits) {
            obj.profits = message.profits.map(e => e ? coin_1.Coin.toAmino(e) : undefined);
        }
        else {
            obj.profits = [];
        }
        obj.number_of_trades = message.numberOfTrades;
        if (message.route) {
            obj.route = message.route.map(e => e.toString());
        }
        else {
            obj.route = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return exports.RouteStatistics.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/route-statistics",
            value: exports.RouteStatistics.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.RouteStatistics.decode(message.value);
    },
    toProto(message) {
        return exports.RouteStatistics.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.RouteStatistics",
            value: exports.RouteStatistics.encode(message).finish()
        };
    }
};
function createBasePoolWeights() {
    return {
        stableWeight: BigInt(0),
        balancerWeight: BigInt(0),
        concentratedWeight: BigInt(0)
    };
}
exports.PoolWeights = {
    typeUrl: "/osmosis.protorev.v1beta1.PoolWeights",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.stableWeight !== BigInt(0)) {
            writer.uint32(8).uint64(message.stableWeight);
        }
        if (message.balancerWeight !== BigInt(0)) {
            writer.uint32(16).uint64(message.balancerWeight);
        }
        if (message.concentratedWeight !== BigInt(0)) {
            writer.uint32(24).uint64(message.concentratedWeight);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePoolWeights();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.stableWeight = reader.uint64();
                    break;
                case 2:
                    message.balancerWeight = reader.uint64();
                    break;
                case 3:
                    message.concentratedWeight = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBasePoolWeights();
        message.stableWeight = object.stableWeight !== undefined && object.stableWeight !== null ? BigInt(object.stableWeight.toString()) : BigInt(0);
        message.balancerWeight = object.balancerWeight !== undefined && object.balancerWeight !== null ? BigInt(object.balancerWeight.toString()) : BigInt(0);
        message.concentratedWeight = object.concentratedWeight !== undefined && object.concentratedWeight !== null ? BigInt(object.concentratedWeight.toString()) : BigInt(0);
        return message;
    },
    fromAmino(object) {
        return {
            stableWeight: BigInt(object.stable_weight),
            balancerWeight: BigInt(object.balancer_weight),
            concentratedWeight: BigInt(object.concentrated_weight)
        };
    },
    toAmino(message) {
        const obj = {};
        obj.stable_weight = message.stableWeight ? message.stableWeight.toString() : undefined;
        obj.balancer_weight = message.balancerWeight ? message.balancerWeight.toString() : undefined;
        obj.concentrated_weight = message.concentratedWeight ? message.concentratedWeight.toString() : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.PoolWeights.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/pool-weights",
            value: exports.PoolWeights.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.PoolWeights.decode(message.value);
    },
    toProto(message) {
        return exports.PoolWeights.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.PoolWeights",
            value: exports.PoolWeights.encode(message).finish()
        };
    }
};
function createBaseBaseDenom() {
    return {
        denom: "",
        stepSize: ""
    };
}
exports.BaseDenom = {
    typeUrl: "/osmosis.protorev.v1beta1.BaseDenom",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.denom !== "") {
            writer.uint32(10).string(message.denom);
        }
        if (message.stepSize !== "") {
            writer.uint32(18).string(message.stepSize);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBaseDenom();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.denom = reader.string();
                    break;
                case 2:
                    message.stepSize = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseBaseDenom();
        message.denom = object.denom ?? "";
        message.stepSize = object.stepSize ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            denom: object.denom,
            stepSize: object.step_size
        };
    },
    toAmino(message) {
        const obj = {};
        obj.denom = message.denom;
        obj.step_size = message.stepSize;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.BaseDenom.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/base-denom",
            value: exports.BaseDenom.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.BaseDenom.decode(message.value);
    },
    toProto(message) {
        return exports.BaseDenom.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.BaseDenom",
            value: exports.BaseDenom.encode(message).finish()
        };
    }
};
