// osmojs/dist/codegen/osmosis/protorev/v1beta1/tx.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSetBaseDenomsResponse = exports.MsgSetBaseDenoms = exports.MsgSetMaxPoolPointsPerBlockResponse = exports.MsgSetMaxPoolPointsPerBlock = exports.MsgSetMaxPoolPointsPerTxResponse = exports.MsgSetMaxPoolPointsPerTx = exports.MsgSetPoolWeightsResponse = exports.MsgSetPoolWeights = exports.MsgSetDeveloperAccountResponse = exports.MsgSetDeveloperAccount = exports.MsgSetHotRoutesResponse = exports.MsgSetHotRoutes = void 0;
const protorev_1 = require("./protorev");
const binary_1 = require("../../../binary");
function createBaseMsgSetHotRoutes() {
    return {
        admin: "",
        hotRoutes: []
    };
}
exports.MsgSetHotRoutes = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetHotRoutes",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.admin !== "") {
            writer.uint32(10).string(message.admin);
        }
        for (const v of message.hotRoutes) {
            protorev_1.TokenPairArbRoutes.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetHotRoutes();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.admin = reader.string();
                    break;
                case 2:
                    message.hotRoutes.push(protorev_1.TokenPairArbRoutes.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgSetHotRoutes();
        message.admin = object.admin ?? "";
        message.hotRoutes = object.hotRoutes?.map(e => protorev_1.TokenPairArbRoutes.fromPartial(e)) || [];
        return message;
    },
    fromAmino(object) {
        return {
            admin: object.admin,
            hotRoutes: Array.isArray(object?.hot_routes) ? object.hot_routes.map((e) => protorev_1.TokenPairArbRoutes.fromAmino(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        obj.admin = message.admin;
        if (message.hotRoutes) {
            obj.hot_routes = message.hotRoutes.map(e => e ? protorev_1.TokenPairArbRoutes.toAmino(e) : undefined);
        }
        else {
            obj.hot_routes = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetHotRoutes.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/MsgSetHotRoutes",
            value: exports.MsgSetHotRoutes.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetHotRoutes.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetHotRoutes.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetHotRoutes",
            value: exports.MsgSetHotRoutes.encode(message).finish()
        };
    }
};
function createBaseMsgSetHotRoutesResponse() {
    return {};
}
exports.MsgSetHotRoutesResponse = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetHotRoutesResponse",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetHotRoutesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetHotRoutesResponse();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetHotRoutesResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-hot-routes-response",
            value: exports.MsgSetHotRoutesResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetHotRoutesResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetHotRoutesResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetHotRoutesResponse",
            value: exports.MsgSetHotRoutesResponse.encode(message).finish()
        };
    }
};
function createBaseMsgSetDeveloperAccount() {
    return {
        admin: "",
        developerAccount: ""
    };
}
exports.MsgSetDeveloperAccount = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetDeveloperAccount",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.admin !== "") {
            writer.uint32(10).string(message.admin);
        }
        if (message.developerAccount !== "") {
            writer.uint32(18).string(message.developerAccount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetDeveloperAccount();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.admin = reader.string();
                    break;
                case 2:
                    message.developerAccount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgSetDeveloperAccount();
        message.admin = object.admin ?? "";
        message.developerAccount = object.developerAccount ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            admin: object.admin,
            developerAccount: object.developer_account
        };
    },
    toAmino(message) {
        const obj = {};
        obj.admin = message.admin;
        obj.developer_account = message.developerAccount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetDeveloperAccount.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/MsgSetDeveloperAccount",
            value: exports.MsgSetDeveloperAccount.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetDeveloperAccount.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetDeveloperAccount.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetDeveloperAccount",
            value: exports.MsgSetDeveloperAccount.encode(message).finish()
        };
    }
};
function createBaseMsgSetDeveloperAccountResponse() {
    return {};
}
exports.MsgSetDeveloperAccountResponse = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetDeveloperAccountResponse",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetDeveloperAccountResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetDeveloperAccountResponse();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetDeveloperAccountResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-developer-account-response",
            value: exports.MsgSetDeveloperAccountResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetDeveloperAccountResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetDeveloperAccountResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetDeveloperAccountResponse",
            value: exports.MsgSetDeveloperAccountResponse.encode(message).finish()
        };
    }
};
function createBaseMsgSetPoolWeights() {
    return {
        admin: "",
        poolWeights: protorev_1.PoolWeights.fromPartial({})
    };
}
exports.MsgSetPoolWeights = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetPoolWeights",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.admin !== "") {
            writer.uint32(10).string(message.admin);
        }
        if (message.poolWeights !== undefined) {
            protorev_1.PoolWeights.encode(message.poolWeights, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetPoolWeights();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.admin = reader.string();
                    break;
                case 2:
                    message.poolWeights = protorev_1.PoolWeights.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgSetPoolWeights();
        message.admin = object.admin ?? "";
        message.poolWeights = object.poolWeights !== undefined && object.poolWeights !== null ? protorev_1.PoolWeights.fromPartial(object.poolWeights) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            admin: object.admin,
            poolWeights: object?.pool_weights ? protorev_1.PoolWeights.fromAmino(object.pool_weights) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.admin = message.admin;
        obj.pool_weights = message.poolWeights ? protorev_1.PoolWeights.toAmino(message.poolWeights) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetPoolWeights.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-pool-weights",
            value: exports.MsgSetPoolWeights.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetPoolWeights.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetPoolWeights.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetPoolWeights",
            value: exports.MsgSetPoolWeights.encode(message).finish()
        };
    }
};
function createBaseMsgSetPoolWeightsResponse() {
    return {};
}
exports.MsgSetPoolWeightsResponse = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetPoolWeightsResponse",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetPoolWeightsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetPoolWeightsResponse();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetPoolWeightsResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-pool-weights-response",
            value: exports.MsgSetPoolWeightsResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetPoolWeightsResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetPoolWeightsResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetPoolWeightsResponse",
            value: exports.MsgSetPoolWeightsResponse.encode(message).finish()
        };
    }
};
function createBaseMsgSetMaxPoolPointsPerTx() {
    return {
        admin: "",
        maxPoolPointsPerTx: BigInt(0)
    };
}
exports.MsgSetMaxPoolPointsPerTx = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerTx",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.admin !== "") {
            writer.uint32(10).string(message.admin);
        }
        if (message.maxPoolPointsPerTx !== BigInt(0)) {
            writer.uint32(16).uint64(message.maxPoolPointsPerTx);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMaxPoolPointsPerTx();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.admin = reader.string();
                    break;
                case 2:
                    message.maxPoolPointsPerTx = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgSetMaxPoolPointsPerTx();
        message.admin = object.admin ?? "";
        message.maxPoolPointsPerTx = object.maxPoolPointsPerTx !== undefined && object.maxPoolPointsPerTx !== null ? BigInt(object.maxPoolPointsPerTx.toString()) : BigInt(0);
        return message;
    },
    fromAmino(object) {
        return {
            admin: object.admin,
            maxPoolPointsPerTx: BigInt(object.max_pool_points_per_tx)
        };
    },
    toAmino(message) {
        const obj = {};
        obj.admin = message.admin;
        obj.max_pool_points_per_tx = message.maxPoolPointsPerTx ? message.maxPoolPointsPerTx.toString() : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetMaxPoolPointsPerTx.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-max-pool-points-per-tx",
            value: exports.MsgSetMaxPoolPointsPerTx.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetMaxPoolPointsPerTx.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetMaxPoolPointsPerTx.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerTx",
            value: exports.MsgSetMaxPoolPointsPerTx.encode(message).finish()
        };
    }
};
function createBaseMsgSetMaxPoolPointsPerTxResponse() {
    return {};
}
exports.MsgSetMaxPoolPointsPerTxResponse = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerTxResponse",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMaxPoolPointsPerTxResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetMaxPoolPointsPerTxResponse();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetMaxPoolPointsPerTxResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-max-pool-points-per-tx-response",
            value: exports.MsgSetMaxPoolPointsPerTxResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetMaxPoolPointsPerTxResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetMaxPoolPointsPerTxResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerTxResponse",
            value: exports.MsgSetMaxPoolPointsPerTxResponse.encode(message).finish()
        };
    }
};
function createBaseMsgSetMaxPoolPointsPerBlock() {
    return {
        admin: "",
        maxPoolPointsPerBlock: BigInt(0)
    };
}
exports.MsgSetMaxPoolPointsPerBlock = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerBlock",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.admin !== "") {
            writer.uint32(10).string(message.admin);
        }
        if (message.maxPoolPointsPerBlock !== BigInt(0)) {
            writer.uint32(16).uint64(message.maxPoolPointsPerBlock);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMaxPoolPointsPerBlock();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.admin = reader.string();
                    break;
                case 2:
                    message.maxPoolPointsPerBlock = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgSetMaxPoolPointsPerBlock();
        message.admin = object.admin ?? "";
        message.maxPoolPointsPerBlock = object.maxPoolPointsPerBlock !== undefined && object.maxPoolPointsPerBlock !== null ? BigInt(object.maxPoolPointsPerBlock.toString()) : BigInt(0);
        return message;
    },
    fromAmino(object) {
        return {
            admin: object.admin,
            maxPoolPointsPerBlock: BigInt(object.max_pool_points_per_block)
        };
    },
    toAmino(message) {
        const obj = {};
        obj.admin = message.admin;
        obj.max_pool_points_per_block = message.maxPoolPointsPerBlock ? message.maxPoolPointsPerBlock.toString() : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetMaxPoolPointsPerBlock.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-max-pool-points-per-block",
            value: exports.MsgSetMaxPoolPointsPerBlock.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetMaxPoolPointsPerBlock.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetMaxPoolPointsPerBlock.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerBlock",
            value: exports.MsgSetMaxPoolPointsPerBlock.encode(message).finish()
        };
    }
};
function createBaseMsgSetMaxPoolPointsPerBlockResponse() {
    return {};
}
exports.MsgSetMaxPoolPointsPerBlockResponse = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerBlockResponse",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetMaxPoolPointsPerBlockResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetMaxPoolPointsPerBlockResponse();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetMaxPoolPointsPerBlockResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-max-pool-points-per-block-response",
            value: exports.MsgSetMaxPoolPointsPerBlockResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetMaxPoolPointsPerBlockResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetMaxPoolPointsPerBlockResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetMaxPoolPointsPerBlockResponse",
            value: exports.MsgSetMaxPoolPointsPerBlockResponse.encode(message).finish()
        };
    }
};
function createBaseMsgSetBaseDenoms() {
    return {
        admin: "",
        baseDenoms: []
    };
}
exports.MsgSetBaseDenoms = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetBaseDenoms",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.admin !== "") {
            writer.uint32(10).string(message.admin);
        }
        for (const v of message.baseDenoms) {
            protorev_1.BaseDenom.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetBaseDenoms();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.admin = reader.string();
                    break;
                case 2:
                    message.baseDenoms.push(protorev_1.BaseDenom.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseMsgSetBaseDenoms();
        message.admin = object.admin ?? "";
        message.baseDenoms = object.baseDenoms?.map(e => protorev_1.BaseDenom.fromPartial(e)) || [];
        return message;
    },
    fromAmino(object) {
        return {
            admin: object.admin,
            baseDenoms: Array.isArray(object?.base_denoms) ? object.base_denoms.map((e) => protorev_1.BaseDenom.fromAmino(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        obj.admin = message.admin;
        if (message.baseDenoms) {
            obj.base_denoms = message.baseDenoms.map(e => e ? protorev_1.BaseDenom.toAmino(e) : undefined);
        }
        else {
            obj.base_denoms = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetBaseDenoms.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-base-denoms",
            value: exports.MsgSetBaseDenoms.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetBaseDenoms.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetBaseDenoms.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetBaseDenoms",
            value: exports.MsgSetBaseDenoms.encode(message).finish()
        };
    }
};
function createBaseMsgSetBaseDenomsResponse() {
    return {};
}
exports.MsgSetBaseDenomsResponse = {
    typeUrl: "/osmosis.protorev.v1beta1.MsgSetBaseDenomsResponse",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMsgSetBaseDenomsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseMsgSetBaseDenomsResponse();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.MsgSetBaseDenomsResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/protorev/set-base-denoms-response",
            value: exports.MsgSetBaseDenomsResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.MsgSetBaseDenomsResponse.decode(message.value);
    },
    toProto(message) {
        return exports.MsgSetBaseDenomsResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.protorev.v1beta1.MsgSetBaseDenomsResponse",
            value: exports.MsgSetBaseDenomsResponse.encode(message).finish()
        };
    }
};
