// osmojs/dist/codegen/osmosis/protorev/v1beta1/tx.rpc.msg.mjs
import { BinaryReader } from "../../../binary";
import { MsgSetHotRoutes, MsgSetHotRoutesResponse, MsgSetDeveloperAccount, MsgSetDeveloperAccountResponse, MsgSetMaxPoolPointsPerTx, MsgSetMaxPoolPointsPerTxResponse, MsgSetMaxPoolPointsPerBlock, MsgSetMaxPoolPointsPerBlockResponse, MsgSetPoolWeights, MsgSetPoolWeightsResponse, MsgSetBaseDenoms, MsgSetBaseDenomsResponse } from "./tx";
export class MsgClientImpl {
    rpc;
    constructor(rpc) {
        this.rpc = rpc;
        this.setHotRoutes = this.setHotRoutes.bind(this);
        this.setDeveloperAccount = this.setDeveloperAccount.bind(this);
        this.setMaxPoolPointsPerTx = this.setMaxPoolPointsPerTx.bind(this);
        this.setMaxPoolPointsPerBlock = this.setMaxPoolPointsPerBlock.bind(this);
        this.setPoolWeights = this.setPoolWeights.bind(this);
        this.setBaseDenoms = this.setBaseDenoms.bind(this);
    }
    setHotRoutes(request) {
        const data = MsgSetHotRoutes.encode(request).finish();
        const promise = this.rpc.request("osmosis.protorev.v1beta1.Msg", "SetHotRoutes", data);
        return promise.then(data => MsgSetHotRoutesResponse.decode(new BinaryReader(data)));
    }
    setDeveloperAccount(request) {
        const data = MsgSetDeveloperAccount.encode(request).finish();
        const promise = this.rpc.request("osmosis.protorev.v1beta1.Msg", "SetDeveloperAccount", data);
        return promise.then(data => MsgSetDeveloperAccountResponse.decode(new BinaryReader(data)));
    }
    setMaxPoolPointsPerTx(request) {
        const data = MsgSetMaxPoolPointsPerTx.encode(request).finish();
        const promise = this.rpc.request("osmosis.protorev.v1beta1.Msg", "SetMaxPoolPointsPerTx", data);
        return promise.then(data => MsgSetMaxPoolPointsPerTxResponse.decode(new BinaryReader(data)));
    }
    setMaxPoolPointsPerBlock(request) {
        const data = MsgSetMaxPoolPointsPerBlock.encode(request).finish();
        const promise = this.rpc.request("osmosis.protorev.v1beta1.Msg", "SetMaxPoolPointsPerBlock", data);
        return promise.then(data => MsgSetMaxPoolPointsPerBlockResponse.decode(new BinaryReader(data)));
    }
    setPoolWeights(request) {
        const data = MsgSetPoolWeights.encode(request).finish();
        const promise = this.rpc.request("osmosis.protorev.v1beta1.Msg", "SetPoolWeights", data);
        return promise.then(data => MsgSetPoolWeightsResponse.decode(new BinaryReader(data)));
    }
    setBaseDenoms(request) {
        const data = MsgSetBaseDenoms.encode(request).finish();
        const promise = this.rpc.request("osmosis.protorev.v1beta1.Msg", "SetBaseDenoms", data);
        return promise.then(data => MsgSetBaseDenomsResponse.decode(new BinaryReader(data)));
    }
}
