// osmojs/dist/codegen/osmosis/poolmanager/v1beta1/tx.rpc.msg.mjs
import { BinaryReader } from "../../../binary";
import { MsgSwapExactAmountIn, MsgSwapExactAmountInResponse, MsgSwapExactAmountOut, MsgSwapExactAmountOutResponse, MsgSplitRouteSwapExactAmountIn, MsgSplitRouteSwapExactAmountInResponse, MsgSplitRouteSwapExactAmountOut, MsgSplitRouteSwapExactAmountOutResponse } from "./tx";
export class MsgClientImpl {
    rpc;
    constructor(rpc) {
        this.rpc = rpc;
        this.swapExactAmountIn = this.swapExactAmountIn.bind(this);
        this.swapExactAmountOut = this.swapExactAmountOut.bind(this);
        this.splitRouteSwapExactAmountIn = this.splitRouteSwapExactAmountIn.bind(this);
        this.splitRouteSwapExactAmountOut = this.splitRouteSwapExactAmountOut.bind(this);
    }
    swapExactAmountIn(request) {
        const data = MsgSwapExactAmountIn.encode(request).finish();
        const promise = this.rpc.request("osmosis.poolmanager.v1beta1.Msg", "SwapExactAmountIn", data);
        return promise.then(data => MsgSwapExactAmountInResponse.decode(new BinaryReader(data)));
    }
    swapExactAmountOut(request) {
        const data = MsgSwapExactAmountOut.encode(request).finish();
        const promise = this.rpc.request("osmosis.poolmanager.v1beta1.Msg", "SwapExactAmountOut", data);
        return promise.then(data => MsgSwapExactAmountOutResponse.decode(new BinaryReader(data)));
    }
    splitRouteSwapExactAmountIn(request) {
        const data = MsgSplitRouteSwapExactAmountIn.encode(request).finish();
        const promise = this.rpc.request("osmosis.poolmanager.v1beta1.Msg", "SplitRouteSwapExactAmountIn", data);
        return promise.then(data => MsgSplitRouteSwapExactAmountInResponse.decode(new BinaryReader(data)));
    }
    splitRouteSwapExactAmountOut(request) {
        const data = MsgSplitRouteSwapExactAmountOut.encode(request).finish();
        const promise = this.rpc.request("osmosis.poolmanager.v1beta1.Msg", "SplitRouteSwapExactAmountOut", data);
        return promise.then(data => MsgSplitRouteSwapExactAmountOutResponse.decode(new BinaryReader(data)));
    }
}
