// osmojs/dist/codegen/osmosis/cosmwasmpool/v1beta1/model/module_sudo_msg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SwapExactAmountOutSudoMsgResponse = exports.SwapExactAmountOutSudoMsg = exports.SwapExactAmountOut = exports.SwapExactAmountInSudoMsgResponse = exports.SwapExactAmountInSudoMsg = exports.SwapExactAmountIn = void 0;
const coin_1 = require("../../../../cosmos/base/v1beta1/coin");
const binary_1 = require("../../../../binary");
const math_1 = require("@cosmjs/math");
function createBaseSwapExactAmountIn() {
    return {
        sender: "",
        tokenIn: undefined,
        tokenOutDenom: "",
        tokenOutMinAmount: "",
        swapFee: ""
    };
}
exports.SwapExactAmountIn = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountIn",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.tokenIn !== undefined) {
            coin_1.Coin.encode(message.tokenIn, writer.uint32(18).fork()).ldelim();
        }
        if (message.tokenOutDenom !== "") {
            writer.uint32(26).string(message.tokenOutDenom);
        }
        if (message.tokenOutMinAmount !== "") {
            writer.uint32(34).string(message.tokenOutMinAmount);
        }
        if (message.swapFee !== "") {
            writer.uint32(42).string(math_1.Decimal.fromUserInput(message.swapFee, 18).atomics);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSwapExactAmountIn();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.string();
                    break;
                case 2:
                    message.tokenIn = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.tokenOutDenom = reader.string();
                    break;
                case 4:
                    message.tokenOutMinAmount = reader.string();
                    break;
                case 5:
                    message.swapFee = math_1.Decimal.fromAtomics(reader.string(), 18).toString();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSwapExactAmountIn();
        message.sender = object.sender ?? "";
        message.tokenIn = object.tokenIn !== undefined && object.tokenIn !== null ? coin_1.Coin.fromPartial(object.tokenIn) : undefined;
        message.tokenOutDenom = object.tokenOutDenom ?? "";
        message.tokenOutMinAmount = object.tokenOutMinAmount ?? "";
        message.swapFee = object.swapFee ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            sender: object.sender,
            tokenIn: object?.token_in ? coin_1.Coin.fromAmino(object.token_in) : undefined,
            tokenOutDenom: object.token_out_denom,
            tokenOutMinAmount: object.token_out_min_amount,
            swapFee: object.swap_fee
        };
    },
    toAmino(message) {
        const obj = {};
        obj.sender = message.sender;
        obj.token_in = message.tokenIn ? coin_1.Coin.toAmino(message.tokenIn) : undefined;
        obj.token_out_denom = message.tokenOutDenom;
        obj.token_out_min_amount = message.tokenOutMinAmount;
        obj.swap_fee = message.swapFee;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SwapExactAmountIn.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/swap-exact-amount-in",
            value: exports.SwapExactAmountIn.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SwapExactAmountIn.decode(message.value);
    },
    toProto(message) {
        return exports.SwapExactAmountIn.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountIn",
            value: exports.SwapExactAmountIn.encode(message).finish()
        };
    }
};
function createBaseSwapExactAmountInSudoMsg() {
    return {
        swapExactAmountIn: exports.SwapExactAmountIn.fromPartial({})
    };
}
exports.SwapExactAmountInSudoMsg = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountInSudoMsg",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.swapExactAmountIn !== undefined) {
            exports.SwapExactAmountIn.encode(message.swapExactAmountIn, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSwapExactAmountInSudoMsg();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.swapExactAmountIn = exports.SwapExactAmountIn.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSwapExactAmountInSudoMsg();
        message.swapExactAmountIn = object.swapExactAmountIn !== undefined && object.swapExactAmountIn !== null ? exports.SwapExactAmountIn.fromPartial(object.swapExactAmountIn) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            swapExactAmountIn: object?.swap_exact_amount_in ? exports.SwapExactAmountIn.fromAmino(object.swap_exact_amount_in) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.swap_exact_amount_in = message.swapExactAmountIn ? exports.SwapExactAmountIn.toAmino(message.swapExactAmountIn) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SwapExactAmountInSudoMsg.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/swap-exact-amount-in-sudo-msg",
            value: exports.SwapExactAmountInSudoMsg.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SwapExactAmountInSudoMsg.decode(message.value);
    },
    toProto(message) {
        return exports.SwapExactAmountInSudoMsg.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountInSudoMsg",
            value: exports.SwapExactAmountInSudoMsg.encode(message).finish()
        };
    }
};
function createBaseSwapExactAmountInSudoMsgResponse() {
    return {
        tokenOutAmount: ""
    };
}
exports.SwapExactAmountInSudoMsgResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountInSudoMsgResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.tokenOutAmount !== "") {
            writer.uint32(10).string(message.tokenOutAmount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSwapExactAmountInSudoMsgResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tokenOutAmount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSwapExactAmountInSudoMsgResponse();
        message.tokenOutAmount = object.tokenOutAmount ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            tokenOutAmount: object.token_out_amount
        };
    },
    toAmino(message) {
        const obj = {};
        obj.token_out_amount = message.tokenOutAmount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SwapExactAmountInSudoMsgResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/swap-exact-amount-in-sudo-msg-response",
            value: exports.SwapExactAmountInSudoMsgResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SwapExactAmountInSudoMsgResponse.decode(message.value);
    },
    toProto(message) {
        return exports.SwapExactAmountInSudoMsgResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountInSudoMsgResponse",
            value: exports.SwapExactAmountInSudoMsgResponse.encode(message).finish()
        };
    }
};
function createBaseSwapExactAmountOut() {
    return {
        sender: "",
        tokenOut: undefined,
        tokenInDenom: "",
        tokenInMaxAmount: "",
        swapFee: ""
    };
}
exports.SwapExactAmountOut = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountOut",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.tokenOut !== undefined) {
            coin_1.Coin.encode(message.tokenOut, writer.uint32(18).fork()).ldelim();
        }
        if (message.tokenInDenom !== "") {
            writer.uint32(26).string(message.tokenInDenom);
        }
        if (message.tokenInMaxAmount !== "") {
            writer.uint32(34).string(message.tokenInMaxAmount);
        }
        if (message.swapFee !== "") {
            writer.uint32(42).string(math_1.Decimal.fromUserInput(message.swapFee, 18).atomics);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSwapExactAmountOut();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.string();
                    break;
                case 2:
                    message.tokenOut = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.tokenInDenom = reader.string();
                    break;
                case 4:
                    message.tokenInMaxAmount = reader.string();
                    break;
                case 5:
                    message.swapFee = math_1.Decimal.fromAtomics(reader.string(), 18).toString();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSwapExactAmountOut();
        message.sender = object.sender ?? "";
        message.tokenOut = object.tokenOut !== undefined && object.tokenOut !== null ? coin_1.Coin.fromPartial(object.tokenOut) : undefined;
        message.tokenInDenom = object.tokenInDenom ?? "";
        message.tokenInMaxAmount = object.tokenInMaxAmount ?? "";
        message.swapFee = object.swapFee ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            sender: object.sender,
            tokenOut: object?.token_out ? coin_1.Coin.fromAmino(object.token_out) : undefined,
            tokenInDenom: object.token_in_denom,
            tokenInMaxAmount: object.token_in_max_amount,
            swapFee: object.swap_fee
        };
    },
    toAmino(message) {
        const obj = {};
        obj.sender = message.sender;
        obj.token_out = message.tokenOut ? coin_1.Coin.toAmino(message.tokenOut) : undefined;
        obj.token_in_denom = message.tokenInDenom;
        obj.token_in_max_amount = message.tokenInMaxAmount;
        obj.swap_fee = message.swapFee;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SwapExactAmountOut.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/swap-exact-amount-out",
            value: exports.SwapExactAmountOut.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SwapExactAmountOut.decode(message.value);
    },
    toProto(message) {
        return exports.SwapExactAmountOut.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountOut",
            value: exports.SwapExactAmountOut.encode(message).finish()
        };
    }
};
function createBaseSwapExactAmountOutSudoMsg() {
    return {
        swapExactAmountOut: exports.SwapExactAmountOut.fromPartial({})
    };
}
exports.SwapExactAmountOutSudoMsg = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountOutSudoMsg",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.swapExactAmountOut !== undefined) {
            exports.SwapExactAmountOut.encode(message.swapExactAmountOut, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSwapExactAmountOutSudoMsg();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.swapExactAmountOut = exports.SwapExactAmountOut.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSwapExactAmountOutSudoMsg();
        message.swapExactAmountOut = object.swapExactAmountOut !== undefined && object.swapExactAmountOut !== null ? exports.SwapExactAmountOut.fromPartial(object.swapExactAmountOut) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            swapExactAmountOut: object?.swap_exact_amount_out ? exports.SwapExactAmountOut.fromAmino(object.swap_exact_amount_out) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.swap_exact_amount_out = message.swapExactAmountOut ? exports.SwapExactAmountOut.toAmino(message.swapExactAmountOut) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SwapExactAmountOutSudoMsg.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/swap-exact-amount-out-sudo-msg",
            value: exports.SwapExactAmountOutSudoMsg.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SwapExactAmountOutSudoMsg.decode(message.value);
    },
    toProto(message) {
        return exports.SwapExactAmountOutSudoMsg.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountOutSudoMsg",
            value: exports.SwapExactAmountOutSudoMsg.encode(message).finish()
        };
    }
};
function createBaseSwapExactAmountOutSudoMsgResponse() {
    return {
        tokenInAmount: ""
    };
}
exports.SwapExactAmountOutSudoMsgResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountOutSudoMsgResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.tokenInAmount !== "") {
            writer.uint32(10).string(message.tokenInAmount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSwapExactAmountOutSudoMsgResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tokenInAmount = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSwapExactAmountOutSudoMsgResponse();
        message.tokenInAmount = object.tokenInAmount ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            tokenInAmount: object.token_in_amount
        };
    },
    toAmino(message) {
        const obj = {};
        obj.token_in_amount = message.tokenInAmount;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SwapExactAmountOutSudoMsgResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/swap-exact-amount-out-sudo-msg-response",
            value: exports.SwapExactAmountOutSudoMsgResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SwapExactAmountOutSudoMsgResponse.decode(message.value);
    },
    toProto(message) {
        return exports.SwapExactAmountOutSudoMsgResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SwapExactAmountOutSudoMsgResponse",
            value: exports.SwapExactAmountOutSudoMsgResponse.encode(message).finish()
        };
    }
};
