// osmojs/dist/codegen/osmosis/cosmwasmpool/v1beta1/model/module_query_msg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalcInAmtGivenOutResponse = exports.CalcInAmtGivenOutRequest = exports.CalcInAmtGivenOut = exports.CalcOutAmtGivenInResponse = exports.CalcOutAmtGivenInRequest = exports.CalcOutAmtGivenIn = void 0;
const coin_1 = require("../../../../cosmos/base/v1beta1/coin");
const binary_1 = require("../../../../binary");
const math_1 = require("@cosmjs/math");
function createBaseCalcOutAmtGivenIn() {
    return {
        tokenIn: undefined,
        tokenOutDenom: "",
        swapFee: ""
    };
}
exports.CalcOutAmtGivenIn = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcOutAmtGivenIn",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.tokenIn !== undefined) {
            coin_1.Coin.encode(message.tokenIn, writer.uint32(10).fork()).ldelim();
        }
        if (message.tokenOutDenom !== "") {
            writer.uint32(18).string(message.tokenOutDenom);
        }
        if (message.swapFee !== "") {
            writer.uint32(26).string(math_1.Decimal.fromUserInput(message.swapFee, 18).atomics);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCalcOutAmtGivenIn();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tokenIn = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.tokenOutDenom = reader.string();
                    break;
                case 3:
                    message.swapFee = math_1.Decimal.fromAtomics(reader.string(), 18).toString();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseCalcOutAmtGivenIn();
        message.tokenIn = object.tokenIn !== undefined && object.tokenIn !== null ? coin_1.Coin.fromPartial(object.tokenIn) : undefined;
        message.tokenOutDenom = object.tokenOutDenom ?? "";
        message.swapFee = object.swapFee ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            tokenIn: object?.token_in ? coin_1.Coin.fromAmino(object.token_in) : undefined,
            tokenOutDenom: object.token_out_denom,
            swapFee: object.swap_fee
        };
    },
    toAmino(message) {
        const obj = {};
        obj.token_in = message.tokenIn ? coin_1.Coin.toAmino(message.tokenIn) : undefined;
        obj.token_out_denom = message.tokenOutDenom;
        obj.swap_fee = message.swapFee;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.CalcOutAmtGivenIn.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/calc-out-amt-given-in",
            value: exports.CalcOutAmtGivenIn.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.CalcOutAmtGivenIn.decode(message.value);
    },
    toProto(message) {
        return exports.CalcOutAmtGivenIn.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcOutAmtGivenIn",
            value: exports.CalcOutAmtGivenIn.encode(message).finish()
        };
    }
};
function createBaseCalcOutAmtGivenInRequest() {
    return {
        calcOutAmtGivenIn: exports.CalcOutAmtGivenIn.fromPartial({})
    };
}
exports.CalcOutAmtGivenInRequest = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcOutAmtGivenInRequest",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.calcOutAmtGivenIn !== undefined) {
            exports.CalcOutAmtGivenIn.encode(message.calcOutAmtGivenIn, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCalcOutAmtGivenInRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.calcOutAmtGivenIn = exports.CalcOutAmtGivenIn.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseCalcOutAmtGivenInRequest();
        message.calcOutAmtGivenIn = object.calcOutAmtGivenIn !== undefined && object.calcOutAmtGivenIn !== null ? exports.CalcOutAmtGivenIn.fromPartial(object.calcOutAmtGivenIn) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            calcOutAmtGivenIn: object?.calc_out_amt_given_in ? exports.CalcOutAmtGivenIn.fromAmino(object.calc_out_amt_given_in) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.calc_out_amt_given_in = message.calcOutAmtGivenIn ? exports.CalcOutAmtGivenIn.toAmino(message.calcOutAmtGivenIn) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.CalcOutAmtGivenInRequest.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/calc-out-amt-given-in-request",
            value: exports.CalcOutAmtGivenInRequest.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.CalcOutAmtGivenInRequest.decode(message.value);
    },
    toProto(message) {
        return exports.CalcOutAmtGivenInRequest.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcOutAmtGivenInRequest",
            value: exports.CalcOutAmtGivenInRequest.encode(message).finish()
        };
    }
};
function createBaseCalcOutAmtGivenInResponse() {
    return {
        tokenOut: undefined
    };
}
exports.CalcOutAmtGivenInResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcOutAmtGivenInResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.tokenOut !== undefined) {
            coin_1.Coin.encode(message.tokenOut, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCalcOutAmtGivenInResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tokenOut = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseCalcOutAmtGivenInResponse();
        message.tokenOut = object.tokenOut !== undefined && object.tokenOut !== null ? coin_1.Coin.fromPartial(object.tokenOut) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            tokenOut: object?.token_out ? coin_1.Coin.fromAmino(object.token_out) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.token_out = message.tokenOut ? coin_1.Coin.toAmino(message.tokenOut) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.CalcOutAmtGivenInResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/calc-out-amt-given-in-response",
            value: exports.CalcOutAmtGivenInResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.CalcOutAmtGivenInResponse.decode(message.value);
    },
    toProto(message) {
        return exports.CalcOutAmtGivenInResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcOutAmtGivenInResponse",
            value: exports.CalcOutAmtGivenInResponse.encode(message).finish()
        };
    }
};
function createBaseCalcInAmtGivenOut() {
    return {
        tokenOut: undefined,
        tokenInDenom: "",
        swapFee: ""
    };
}
exports.CalcInAmtGivenOut = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcInAmtGivenOut",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.tokenOut !== undefined) {
            coin_1.Coin.encode(message.tokenOut, writer.uint32(10).fork()).ldelim();
        }
        if (message.tokenInDenom !== "") {
            writer.uint32(18).string(message.tokenInDenom);
        }
        if (message.swapFee !== "") {
            writer.uint32(26).string(math_1.Decimal.fromUserInput(message.swapFee, 18).atomics);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCalcInAmtGivenOut();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tokenOut = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.tokenInDenom = reader.string();
                    break;
                case 3:
                    message.swapFee = math_1.Decimal.fromAtomics(reader.string(), 18).toString();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseCalcInAmtGivenOut();
        message.tokenOut = object.tokenOut !== undefined && object.tokenOut !== null ? coin_1.Coin.fromPartial(object.tokenOut) : undefined;
        message.tokenInDenom = object.tokenInDenom ?? "";
        message.swapFee = object.swapFee ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            tokenOut: object?.token_out ? coin_1.Coin.fromAmino(object.token_out) : undefined,
            tokenInDenom: object.token_in_denom,
            swapFee: object.swap_fee
        };
    },
    toAmino(message) {
        const obj = {};
        obj.token_out = message.tokenOut ? coin_1.Coin.toAmino(message.tokenOut) : undefined;
        obj.token_in_denom = message.tokenInDenom;
        obj.swap_fee = message.swapFee;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.CalcInAmtGivenOut.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/calc-in-amt-given-out",
            value: exports.CalcInAmtGivenOut.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.CalcInAmtGivenOut.decode(message.value);
    },
    toProto(message) {
        return exports.CalcInAmtGivenOut.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcInAmtGivenOut",
            value: exports.CalcInAmtGivenOut.encode(message).finish()
        };
    }
};
function createBaseCalcInAmtGivenOutRequest() {
    return {
        calcInAmtGivenOut: exports.CalcInAmtGivenOut.fromPartial({})
    };
}
exports.CalcInAmtGivenOutRequest = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcInAmtGivenOutRequest",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.calcInAmtGivenOut !== undefined) {
            exports.CalcInAmtGivenOut.encode(message.calcInAmtGivenOut, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCalcInAmtGivenOutRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.calcInAmtGivenOut = exports.CalcInAmtGivenOut.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseCalcInAmtGivenOutRequest();
        message.calcInAmtGivenOut = object.calcInAmtGivenOut !== undefined && object.calcInAmtGivenOut !== null ? exports.CalcInAmtGivenOut.fromPartial(object.calcInAmtGivenOut) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            calcInAmtGivenOut: object?.calc_in_amt_given_out ? exports.CalcInAmtGivenOut.fromAmino(object.calc_in_amt_given_out) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.calc_in_amt_given_out = message.calcInAmtGivenOut ? exports.CalcInAmtGivenOut.toAmino(message.calcInAmtGivenOut) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.CalcInAmtGivenOutRequest.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/calc-in-amt-given-out-request",
            value: exports.CalcInAmtGivenOutRequest.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.CalcInAmtGivenOutRequest.decode(message.value);
    },
    toProto(message) {
        return exports.CalcInAmtGivenOutRequest.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcInAmtGivenOutRequest",
            value: exports.CalcInAmtGivenOutRequest.encode(message).finish()
        };
    }
};
function createBaseCalcInAmtGivenOutResponse() {
    return {
        tokenIn: undefined
    };
}
exports.CalcInAmtGivenOutResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcInAmtGivenOutResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.tokenIn !== undefined) {
            coin_1.Coin.encode(message.tokenIn, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCalcInAmtGivenOutResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.tokenIn = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseCalcInAmtGivenOutResponse();
        message.tokenIn = object.tokenIn !== undefined && object.tokenIn !== null ? coin_1.Coin.fromPartial(object.tokenIn) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            tokenIn: object?.token_in ? coin_1.Coin.fromAmino(object.token_in) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.token_in = message.tokenIn ? coin_1.Coin.toAmino(message.tokenIn) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.CalcInAmtGivenOutResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/calc-in-amt-given-out-response",
            value: exports.CalcInAmtGivenOutResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.CalcInAmtGivenOutResponse.decode(message.value);
    },
    toProto(message) {
        return exports.CalcInAmtGivenOutResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.CalcInAmtGivenOutResponse",
            value: exports.CalcInAmtGivenOutResponse.encode(message).finish()
        };
    }
};
