// osmojs/dist/codegen/osmosis/cosmwasmpool/v1beta1/model/pool_query_msg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTotalSharesQueryMsgResponse = exports.GetTotalSharesQueryMsg = exports.GetTotalPoolLiquidityQueryMsgResponse = exports.GetTotalPoolLiquidityQueryMsg = exports.EmptyStruct = exports.SpotPriceQueryMsgResponse = exports.SpotPriceQueryMsg = exports.SpotPrice = exports.GetSwapFeeQueryMsgResponse = exports.GetSwapFeeQueryMsg = void 0;
const coin_1 = require("../../../../cosmos/base/v1beta1/coin");
const binary_1 = require("../../../../binary");
const math_1 = require("@cosmjs/math");
function createBaseGetSwapFeeQueryMsg() {
    return {
        getSwapFee: exports.EmptyStruct.fromPartial({})
    };
}
exports.GetSwapFeeQueryMsg = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetSwapFeeQueryMsg",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.getSwapFee !== undefined) {
            exports.EmptyStruct.encode(message.getSwapFee, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetSwapFeeQueryMsg();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.getSwapFee = exports.EmptyStruct.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseGetSwapFeeQueryMsg();
        message.getSwapFee = object.getSwapFee !== undefined && object.getSwapFee !== null ? exports.EmptyStruct.fromPartial(object.getSwapFee) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            getSwapFee: object?.get_swap_fee ? exports.EmptyStruct.fromAmino(object.get_swap_fee) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.get_swap_fee = message.getSwapFee ? exports.EmptyStruct.toAmino(message.getSwapFee) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.GetSwapFeeQueryMsg.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/get-swap-fee-query-msg",
            value: exports.GetSwapFeeQueryMsg.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.GetSwapFeeQueryMsg.decode(message.value);
    },
    toProto(message) {
        return exports.GetSwapFeeQueryMsg.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetSwapFeeQueryMsg",
            value: exports.GetSwapFeeQueryMsg.encode(message).finish()
        };
    }
};
function createBaseGetSwapFeeQueryMsgResponse() {
    return {
        swapFee: ""
    };
}
exports.GetSwapFeeQueryMsgResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetSwapFeeQueryMsgResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.swapFee !== "") {
            writer.uint32(26).string(math_1.Decimal.fromUserInput(message.swapFee, 18).atomics);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetSwapFeeQueryMsgResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 3:
                    message.swapFee = math_1.Decimal.fromAtomics(reader.string(), 18).toString();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseGetSwapFeeQueryMsgResponse();
        message.swapFee = object.swapFee ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            swapFee: object.swap_fee
        };
    },
    toAmino(message) {
        const obj = {};
        obj.swap_fee = message.swapFee;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.GetSwapFeeQueryMsgResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/get-swap-fee-query-msg-response",
            value: exports.GetSwapFeeQueryMsgResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.GetSwapFeeQueryMsgResponse.decode(message.value);
    },
    toProto(message) {
        return exports.GetSwapFeeQueryMsgResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetSwapFeeQueryMsgResponse",
            value: exports.GetSwapFeeQueryMsgResponse.encode(message).finish()
        };
    }
};
function createBaseSpotPrice() {
    return {
        quoteAssetDenom: "",
        baseAssetDenom: ""
    };
}
exports.SpotPrice = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SpotPrice",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.quoteAssetDenom !== "") {
            writer.uint32(10).string(message.quoteAssetDenom);
        }
        if (message.baseAssetDenom !== "") {
            writer.uint32(18).string(message.baseAssetDenom);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSpotPrice();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.quoteAssetDenom = reader.string();
                    break;
                case 2:
                    message.baseAssetDenom = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSpotPrice();
        message.quoteAssetDenom = object.quoteAssetDenom ?? "";
        message.baseAssetDenom = object.baseAssetDenom ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            quoteAssetDenom: object.quote_asset_denom,
            baseAssetDenom: object.base_asset_denom
        };
    },
    toAmino(message) {
        const obj = {};
        obj.quote_asset_denom = message.quoteAssetDenom;
        obj.base_asset_denom = message.baseAssetDenom;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SpotPrice.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/spot-price",
            value: exports.SpotPrice.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SpotPrice.decode(message.value);
    },
    toProto(message) {
        return exports.SpotPrice.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SpotPrice",
            value: exports.SpotPrice.encode(message).finish()
        };
    }
};
function createBaseSpotPriceQueryMsg() {
    return {
        spotPrice: exports.SpotPrice.fromPartial({})
    };
}
exports.SpotPriceQueryMsg = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SpotPriceQueryMsg",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.spotPrice !== undefined) {
            exports.SpotPrice.encode(message.spotPrice, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSpotPriceQueryMsg();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.spotPrice = exports.SpotPrice.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSpotPriceQueryMsg();
        message.spotPrice = object.spotPrice !== undefined && object.spotPrice !== null ? exports.SpotPrice.fromPartial(object.spotPrice) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            spotPrice: object?.spot_price ? exports.SpotPrice.fromAmino(object.spot_price) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.spot_price = message.spotPrice ? exports.SpotPrice.toAmino(message.spotPrice) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SpotPriceQueryMsg.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/spot-price-query-msg",
            value: exports.SpotPriceQueryMsg.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SpotPriceQueryMsg.decode(message.value);
    },
    toProto(message) {
        return exports.SpotPriceQueryMsg.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SpotPriceQueryMsg",
            value: exports.SpotPriceQueryMsg.encode(message).finish()
        };
    }
};
function createBaseSpotPriceQueryMsgResponse() {
    return {
        spotPrice: ""
    };
}
exports.SpotPriceQueryMsgResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.SpotPriceQueryMsgResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.spotPrice !== "") {
            writer.uint32(10).string(message.spotPrice);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSpotPriceQueryMsgResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.spotPrice = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseSpotPriceQueryMsgResponse();
        message.spotPrice = object.spotPrice ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            spotPrice: object.spot_price
        };
    },
    toAmino(message) {
        const obj = {};
        obj.spot_price = message.spotPrice;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.SpotPriceQueryMsgResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/spot-price-query-msg-response",
            value: exports.SpotPriceQueryMsgResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.SpotPriceQueryMsgResponse.decode(message.value);
    },
    toProto(message) {
        return exports.SpotPriceQueryMsgResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.SpotPriceQueryMsgResponse",
            value: exports.SpotPriceQueryMsgResponse.encode(message).finish()
        };
    }
};
function createBaseEmptyStruct() {
    return {};
}
exports.EmptyStruct = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.EmptyStruct",
    encode(_, writer = binary_1.BinaryWriter.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEmptyStruct();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(_) {
        const message = createBaseEmptyStruct();
        return message;
    },
    fromAmino(_) {
        return {};
    },
    toAmino(_) {
        const obj = {};
        return obj;
    },
    fromAminoMsg(object) {
        return exports.EmptyStruct.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/empty-struct",
            value: exports.EmptyStruct.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.EmptyStruct.decode(message.value);
    },
    toProto(message) {
        return exports.EmptyStruct.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.EmptyStruct",
            value: exports.EmptyStruct.encode(message).finish()
        };
    }
};
function createBaseGetTotalPoolLiquidityQueryMsg() {
    return {
        getTotalPoolLiquidity: exports.EmptyStruct.fromPartial({})
    };
}
exports.GetTotalPoolLiquidityQueryMsg = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalPoolLiquidityQueryMsg",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.getTotalPoolLiquidity !== undefined) {
            exports.EmptyStruct.encode(message.getTotalPoolLiquidity, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetTotalPoolLiquidityQueryMsg();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.getTotalPoolLiquidity = exports.EmptyStruct.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseGetTotalPoolLiquidityQueryMsg();
        message.getTotalPoolLiquidity = object.getTotalPoolLiquidity !== undefined && object.getTotalPoolLiquidity !== null ? exports.EmptyStruct.fromPartial(object.getTotalPoolLiquidity) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            getTotalPoolLiquidity: object?.get_total_pool_liquidity ? exports.EmptyStruct.fromAmino(object.get_total_pool_liquidity) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.get_total_pool_liquidity = message.getTotalPoolLiquidity ? exports.EmptyStruct.toAmino(message.getTotalPoolLiquidity) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.GetTotalPoolLiquidityQueryMsg.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/get-total-pool-liquidity-query-msg",
            value: exports.GetTotalPoolLiquidityQueryMsg.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.GetTotalPoolLiquidityQueryMsg.decode(message.value);
    },
    toProto(message) {
        return exports.GetTotalPoolLiquidityQueryMsg.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalPoolLiquidityQueryMsg",
            value: exports.GetTotalPoolLiquidityQueryMsg.encode(message).finish()
        };
    }
};
function createBaseGetTotalPoolLiquidityQueryMsgResponse() {
    return {
        totalPoolLiquidity: []
    };
}
exports.GetTotalPoolLiquidityQueryMsgResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalPoolLiquidityQueryMsgResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        for (const v of message.totalPoolLiquidity) {
            coin_1.Coin.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetTotalPoolLiquidityQueryMsgResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.totalPoolLiquidity.push(coin_1.Coin.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseGetTotalPoolLiquidityQueryMsgResponse();
        message.totalPoolLiquidity = object.totalPoolLiquidity?.map(e => coin_1.Coin.fromPartial(e)) || [];
        return message;
    },
    fromAmino(object) {
        return {
            totalPoolLiquidity: Array.isArray(object?.total_pool_liquidity) ? object.total_pool_liquidity.map((e) => coin_1.Coin.fromAmino(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        if (message.totalPoolLiquidity) {
            obj.total_pool_liquidity = message.totalPoolLiquidity.map(e => e ? coin_1.Coin.toAmino(e) : undefined);
        }
        else {
            obj.total_pool_liquidity = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return exports.GetTotalPoolLiquidityQueryMsgResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/get-total-pool-liquidity-query-msg-response",
            value: exports.GetTotalPoolLiquidityQueryMsgResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.GetTotalPoolLiquidityQueryMsgResponse.decode(message.value);
    },
    toProto(message) {
        return exports.GetTotalPoolLiquidityQueryMsgResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalPoolLiquidityQueryMsgResponse",
            value: exports.GetTotalPoolLiquidityQueryMsgResponse.encode(message).finish()
        };
    }
};
function createBaseGetTotalSharesQueryMsg() {
    return {
        getTotalShares: exports.EmptyStruct.fromPartial({})
    };
}
exports.GetTotalSharesQueryMsg = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalSharesQueryMsg",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.getTotalShares !== undefined) {
            exports.EmptyStruct.encode(message.getTotalShares, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetTotalSharesQueryMsg();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.getTotalShares = exports.EmptyStruct.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseGetTotalSharesQueryMsg();
        message.getTotalShares = object.getTotalShares !== undefined && object.getTotalShares !== null ? exports.EmptyStruct.fromPartial(object.getTotalShares) : undefined;
        return message;
    },
    fromAmino(object) {
        return {
            getTotalShares: object?.get_total_shares ? exports.EmptyStruct.fromAmino(object.get_total_shares) : undefined
        };
    },
    toAmino(message) {
        const obj = {};
        obj.get_total_shares = message.getTotalShares ? exports.EmptyStruct.toAmino(message.getTotalShares) : undefined;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.GetTotalSharesQueryMsg.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/get-total-shares-query-msg",
            value: exports.GetTotalSharesQueryMsg.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.GetTotalSharesQueryMsg.decode(message.value);
    },
    toProto(message) {
        return exports.GetTotalSharesQueryMsg.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalSharesQueryMsg",
            value: exports.GetTotalSharesQueryMsg.encode(message).finish()
        };
    }
};
function createBaseGetTotalSharesQueryMsgResponse() {
    return {
        totalShares: ""
    };
}
exports.GetTotalSharesQueryMsgResponse = {
    typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalSharesQueryMsgResponse",
    encode(message, writer = binary_1.BinaryWriter.create()) {
        if (message.totalShares !== "") {
            writer.uint32(10).string(message.totalShares);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof binary_1.BinaryReader ? input : new binary_1.BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetTotalSharesQueryMsgResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.totalShares = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseGetTotalSharesQueryMsgResponse();
        message.totalShares = object.totalShares ?? "";
        return message;
    },
    fromAmino(object) {
        return {
            totalShares: object.total_shares
        };
    },
    toAmino(message) {
        const obj = {};
        obj.total_shares = message.totalShares;
        return obj;
    },
    fromAminoMsg(object) {
        return exports.GetTotalSharesQueryMsgResponse.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/cosmwasmpool/get-total-shares-query-msg-response",
            value: exports.GetTotalSharesQueryMsgResponse.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return exports.GetTotalSharesQueryMsgResponse.decode(message.value);
    },
    toProto(message) {
        return exports.GetTotalSharesQueryMsgResponse.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.cosmwasmpool.v1beta1.GetTotalSharesQueryMsgResponse",
            value: exports.GetTotalSharesQueryMsgResponse.encode(message).finish()
        };
    }
};
