// osmojs/dist/codegen/osmosis/gamm/v1beta1/gov.mjs
import { BalancerToConcentratedPoolLink } from "./shared";
import { BinaryReader, BinaryWriter } from "../../../binary";
function createBaseReplaceMigrationRecordsProposal() {
    return {
        $typeUrl: "/osmosis.gamm.v1beta1.ReplaceMigrationRecordsProposal",
        title: "",
        description: "",
        records: []
    };
}
export const ReplaceMigrationRecordsProposal = {
    typeUrl: "/osmosis.gamm.v1beta1.ReplaceMigrationRecordsProposal",
    encode(message, writer = BinaryWriter.create()) {
        if (message.title !== "") {
            writer.uint32(10).string(message.title);
        }
        if (message.description !== "") {
            writer.uint32(18).string(message.description);
        }
        for (const v of message.records) {
            BalancerToConcentratedPoolLink.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReplaceMigrationRecordsProposal();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.title = reader.string();
                    break;
                case 2:
                    message.description = reader.string();
                    break;
                case 3:
                    message.records.push(BalancerToConcentratedPoolLink.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseReplaceMigrationRecordsProposal();
        message.title = object.title ?? "";
        message.description = object.description ?? "";
        message.records = object.records?.map(e => BalancerToConcentratedPoolLink.fromPartial(e)) || [];
        return message;
    },
    fromAmino(object) {
        return {
            title: object.title,
            description: object.description,
            records: Array.isArray(object?.records) ? object.records.map((e) => BalancerToConcentratedPoolLink.fromAmino(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        obj.title = message.title;
        obj.description = message.description;
        if (message.records) {
            obj.records = message.records.map(e => e ? BalancerToConcentratedPoolLink.toAmino(e) : undefined);
        }
        else {
            obj.records = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return ReplaceMigrationRecordsProposal.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/ReplaceMigrationRecordsProposal",
            value: ReplaceMigrationRecordsProposal.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return ReplaceMigrationRecordsProposal.decode(message.value);
    },
    toProto(message) {
        return ReplaceMigrationRecordsProposal.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.gamm.v1beta1.ReplaceMigrationRecordsProposal",
            value: ReplaceMigrationRecordsProposal.encode(message).finish()
        };
    }
};
function createBaseUpdateMigrationRecordsProposal() {
    return {
        $typeUrl: "/osmosis.gamm.v1beta1.UpdateMigrationRecordsProposal",
        title: "",
        description: "",
        records: []
    };
}
export const UpdateMigrationRecordsProposal = {
    typeUrl: "/osmosis.gamm.v1beta1.UpdateMigrationRecordsProposal",
    encode(message, writer = BinaryWriter.create()) {
        if (message.title !== "") {
            writer.uint32(10).string(message.title);
        }
        if (message.description !== "") {
            writer.uint32(18).string(message.description);
        }
        for (const v of message.records) {
            BalancerToConcentratedPoolLink.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateMigrationRecordsProposal();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.title = reader.string();
                    break;
                case 2:
                    message.description = reader.string();
                    break;
                case 3:
                    message.records.push(BalancerToConcentratedPoolLink.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromPartial(object) {
        const message = createBaseUpdateMigrationRecordsProposal();
        message.title = object.title ?? "";
        message.description = object.description ?? "";
        message.records = object.records?.map(e => BalancerToConcentratedPoolLink.fromPartial(e)) || [];
        return message;
    },
    fromAmino(object) {
        return {
            title: object.title,
            description: object.description,
            records: Array.isArray(object?.records) ? object.records.map((e) => BalancerToConcentratedPoolLink.fromAmino(e)) : []
        };
    },
    toAmino(message) {
        const obj = {};
        obj.title = message.title;
        obj.description = message.description;
        if (message.records) {
            obj.records = message.records.map(e => e ? BalancerToConcentratedPoolLink.toAmino(e) : undefined);
        }
        else {
            obj.records = [];
        }
        return obj;
    },
    fromAminoMsg(object) {
        return UpdateMigrationRecordsProposal.fromAmino(object.value);
    },
    toAminoMsg(message) {
        return {
            type: "osmosis/UpdateMigrationRecordsProposal",
            value: UpdateMigrationRecordsProposal.toAmino(message)
        };
    },
    fromProtoMsg(message) {
        return UpdateMigrationRecordsProposal.decode(message.value);
    },
    toProto(message) {
        return UpdateMigrationRecordsProposal.encode(message).finish();
    },
    toProtoMsg(message) {
        return {
            typeUrl: "/osmosis.gamm.v1beta1.UpdateMigrationRecordsProposal",
            value: UpdateMigrationRecordsProposal.encode(message).finish()
        };
    }
};
