// osmojs/dist/codegen/ibc/applications/fee/v1/tx.registry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageComposer = exports.load = exports.registry = void 0;
const tx_1 = require("./tx");
exports.registry = [["/ibc.applications.fee.v1.MsgRegisterPayee", tx_1.MsgRegisterPayee], ["/ibc.applications.fee.v1.MsgRegisterCounterpartyPayee", tx_1.MsgRegisterCounterpartyPayee], ["/ibc.applications.fee.v1.MsgPayPacketFee", tx_1.MsgPayPacketFee], ["/ibc.applications.fee.v1.MsgPayPacketFeeAsync", tx_1.MsgPayPacketFeeAsync]];
const load = (protoRegistry) => {
    exports.registry.forEach(([typeUrl, mod]) => {
        protoRegistry.register(typeUrl, mod);
    });
};
exports.load = load;
exports.MessageComposer = {
    encoded: {
        registerPayee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgRegisterPayee",
                value: tx_1.MsgRegisterPayee.encode(value).finish()
            };
        },
        registerCounterpartyPayee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgRegisterCounterpartyPayee",
                value: tx_1.MsgRegisterCounterpartyPayee.encode(value).finish()
            };
        },
        payPacketFee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgPayPacketFee",
                value: tx_1.MsgPayPacketFee.encode(value).finish()
            };
        },
        payPacketFeeAsync(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgPayPacketFeeAsync",
                value: tx_1.MsgPayPacketFeeAsync.encode(value).finish()
            };
        }
    },
    withTypeUrl: {
        registerPayee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgRegisterPayee",
                value
            };
        },
        registerCounterpartyPayee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgRegisterCounterpartyPayee",
                value
            };
        },
        payPacketFee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgPayPacketFee",
                value
            };
        },
        payPacketFeeAsync(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgPayPacketFeeAsync",
                value
            };
        }
    },
    fromPartial: {
        registerPayee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgRegisterPayee",
                value: tx_1.MsgRegisterPayee.fromPartial(value)
            };
        },
        registerCounterpartyPayee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgRegisterCounterpartyPayee",
                value: tx_1.MsgRegisterCounterpartyPayee.fromPartial(value)
            };
        },
        payPacketFee(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgPayPacketFee",
                value: tx_1.MsgPayPacketFee.fromPartial(value)
            };
        },
        payPacketFeeAsync(value) {
            return {
                typeUrl: "/ibc.applications.fee.v1.MsgPayPacketFeeAsync",
                value: tx_1.MsgPayPacketFeeAsync.fromPartial(value)
            };
        }
    }
};
